### PR TITLE
Add Flatten for Try<IEnumerable<IEnumerable<A>>> to Try<IEnumerable<A>>

### PR DESCRIPTION
**Pull request by Merurino**

### CODE REVIEW

File: Bearded.Monads/TryExtensions.cs [hunk 1]
```diff
@@ -192,6 +192,10 @@ public static A ElseThrow<A>(this Try<A> either)
             this Try<Try<A>> ee)
             => ee.SelectMany(id);
 
+        public static Try<IEnumerable<A>> Flatten<A>(
+            this Try<IEnumerable<IEnumerable<A>>> incoming)
+            => incoming.Select(x => x.SelectMany(id));
```
Comment (khanage) [inline Bearded.Monads/TryExtensions.cs]: Why not just flatten?

like this

Comment (Merurino) [reply Bearded.Monads/TryExtensions.cs]: I can't find a way of doing it with any other already defined Flatten in the library.

Comment (khanage) [reply Bearded.Monads/TryExtensions.cs]: Oh! Silly me - would you mind creating an `EnumerableExtensions.cs` file and defining the `Flatten` for `IEnumerable<IEnumerable<A>>` in there?

Comment (khanage) [reply Bearded.Monads/TryExtensions.cs]: The reason I ask is that if there's a specific implementation provided for `Try` that implementation will bleed out into other implementations for other types - better to have it in one spot, no?

Comment (Merurino) [reply Bearded.Monads/TryExtensions.cs]: I agree :) 
Changes done and pushed.

**Pull request merged**